api/indexer/client_test.go

```diff
@@ -59,26 +70,36 @@ func TestIndexerClientBlocks(t *testing.T) {
 
 			client := NewClient(httpServer.URL)
 
-			expectedBlock := executedBlocks[tt.blkHeight]
-			executedBlock, err := client.GetBlockByHeight(ctx, expectedBlock.Block.Hght, parser)
-			if tt.err == nil {
+			executedBlock, err := client.GetBlockByHeight(ctx, tt.blkHeight, parser)
+			if tt.blkHeightErr == nil {
 				r.NoError(err)
+				expectedBlock := executedBlocks[tt.blkHeight-1]
 				r.Equal(expectedBlock.Block, executedBlock.Block)
 			} else {
-				r.ErrorContains(err, tt.err.Error())
+				r.ErrorContains(err, tt.blkHeightErr.Error())
+			}
+
+			expectedBlkID := ids.Empty
+			if tt.blkHeight > 0 {
+				expectedBlock := executedBlocks[tt.blkHeight-1]
+				expectedBlkID = expectedBlock.Block.GetID()
 			}
 
-			executedBlock, err = client.GetBlock(ctx, expectedBlock.Block.GetID(), parser)
-			if tt.err == nil {
+			executedBlock, err = client.GetBlock(ctx, expectedBlkID, parser)
+			if tt.blkHeightErr == nil {
 				r.NoError(err)
-				r.Equal(expectedBlock.Block, executedBlock.Block)
+				r.Equal(executedBlocks[tt.blkHeight-1].Block, executedBlock.Block)
 			} else {
-				r.ErrorContains(err, tt.err.Error())
+				r.ErrorContains(err, tt.blkHeightErr.Error())
 			}
 
 			executedBlock, err = client.GetLatestBlock(ctx, parser)
-			r.NoError(err)
-			r.Equal(executedBlocks[numExecutedBlocks-1].Block, executedBlock.Block)
+			if tt.numExecutedBlocks == 0 {
+				r.ErrorContains(err, database.ErrNotFound.Error())
```

Comment:
Handling the special case this way makes it seem like we default to not including the genesis. Could we include at least the genesis block (we should receive a notification of at least the genesis block on startup)

Reply:
this is not intended to cover the genesis block; it's really about covering the no-blocks-of-any-kind usecaswe.
Adding genesis is ok of course, but I don't feel that it's needed since it wouldn't be treated any differently from the indexer perspective.